src/Orleans.Core/Messaging/ClientMessageCenter.cs

```diff
@@ -137,16 +139,17 @@ internal class ClientMessageCenter : IMessageCenter, IDisposable
                         return gatewayConnections.Values.Count(conn => conn.IsLive);
                     }
                 });
-            if (StatisticsCollector.CollectQueueStats)
+            statisticsLevel = statisticsOptions.Value.CollectionLevel;
+            if (statisticsLevel.CollectQueueStats())
             {
-                queueTracking = new QueueTrackingStatistic("ClientReceiver");
+                queueTracking = new QueueTrackingStatistic("ClientReceiver", statisticsOptions);
```

Comment:
I was more on move CollectQueueStats() method to `QueueTrackingStatistic` itself, since you did that in other StatisticGroup classes. So to keep consistent, we should move `bool CollectXXXStats()` into the `XXXStatisticGroup` class.

But this is fine, it is more like how it was before, as in whether to collect a Stats is in the same place of `CollectionLevel` settings. Only problem is that the pattern is not consistent over all StatisticGroup.  For example, `ApplicationRequestsStatisticsGroup ` has the `CollectXXXStats` method within itself. 

I will leave it to you to decide do you want to keep it consistent or not.